app/Models/User.php

```diff
@@ -951,4 +954,81 @@ public function toSearchableArray(): array
 
         return $array;
     }
+
+    public function getHasActiveOverrideAttribute(): bool
+    {
+        return ! $this->is_active && $this->access_override_until && $this->access_override_until > now();
+    }
+
+    public function getSelfServiceOverrideEligibilityAttribute(): SelfServiceAccessOverrideEligibility
+    {
+        $ineligibleReason = 'Unable to provide a self-service override at this time';
+
+        // phpcs:ignore
+        $INELIGIBLE_NO_FUTURE_DUES_PKG = 'Self-service access overrides are currently unavailable because there are '.
+            'no dues packages with future access end dates';
+        $INELIGIBLE_REQ_CONDS = 'Account and system conditions for self-service override not met';
+        // phpcs:ignore
+        $INELIGIBLE_REQ_TASKS = 'You have outstanding required tasks that must be completed before receiving your '.
+            'self-service override';
+
+        $now = CarbonImmutable::now();
+        $nextAccessEndDuesPkg = DuesPackage::where('access_end', '>', $now)->get()->sortBy('access_end')->first();
+
+        $eligible = false;
+        $userRectifiable = false; // Simple things the user can do (e.g., attend a team meeting, sign the membership
+        // agreement) to become eligible for a self-service access override. As opposed to things the user can't
+        // change such as having paid dues in the past
+        $overrideEndDate = null;
+
+        // conditions
+        $accessNotActive = ! $this->is_access_active;
+        $noExistingOverride = null === $this->access_override_until;
+        $hasNotPaidDues = ! $this->paidDues()->exists();
+        $eligibleDuesPkgExists = null !== $nextAccessEndDuesPkg;
+
+        // tasks
+        $attendedTeamMeeting = $this->attendance()->whereAttendableType('team')->exists();
```

Comment:
Should this exclude training teams?

Reply:
We usually only give out overrides in the context of the actual teams so I suppose yes.  The trainings outside of Apiary aren't really in the context of a team, so I think it's fine to add the restriction and note in documentation but keep the language as is otherwise.

Reply:
I guess also exclude the various versions of core, to be thorough

Reply:
Sounds like we should add a property to teams rather than base this off team name, imo.

Reply:
sgtm

Reply:
On further consideration, I don't think this restriction is worth implementing.  I don't like the idea of adding such a specific property and don't think it's necessarily problematic that people will become eligible for a self-service override from attending a training.

Reply:
Fine with me